### PR TITLE
refactor: so filter logic from message matcher to filter abstract class

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What change does this PR introduce?
I move some of the methods that will be used for conditions as well from message matcher so we can reuse it. I was not able to move as much as I would like to lets see if we can move some more when we are doing the conditions equal usecase.

### CODE REVIEW

File: apps/worker/src/app/workflow/usecases/message-matcher/message-matcher.usecase.spec.ts [hunk 1]
```diff
@@ -1053,7 +1053,7 @@ describe('Message filter matcher', function () {
     it('should add a passed condition', () => {
       const result = MessageMatcher.sumFilters(
         {
-          stepFilters: [],
```
Comment (davidsoderberg) [inline apps/worker/src/app/workflow/usecases/message-matcher/message-matcher.usecase.spec.ts:1056]: Rename this to filters so it is not workflow specific.

File: packages/application-generic/src/utils/filter-processing-details.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
-import { StepFilter } from '@novu/dal';
```
Comment (davidsoderberg) [inline packages/application-generic/src/utils/filter-processing-details.ts:1]: Moved from api to application-generic.

File: packages/application-generic/src/utils/filter.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import * as _ from 'lodash';
```
Comment (davidsoderberg) [inline packages/application-generic/src/utils/filter.ts:1]: Reefactored and moved from Message matcher

File: apps/worker/src/app/workflow/usecases/message-matcher/message-matcher.usecase.ts [hunk 4]
```diff
@@ -153,19 +159,7 @@ export class MessageMatcher {
       type = 'online';
```
Comment (davidsoderberg) [inline apps/worker/src/app/workflow/usecases/message-matcher/message-matcher.usecase.ts:159]: This is step specific so I keep it in the class and just pass it on tho the Filter class.

**Review by ainouzgali — APPROVED** [review]
looks good :)

File: packages/application-generic/src/utils/filter-processing-details.ts [hunk 5]
```diff
@@ -1,7 +1,11 @@
...
-import { ICondition } from '@novu/shared';
+import { StepFilter, SubscriberEntity } from '@novu/dal';
+import { ICondition, ITriggerPayload } from '@novu/shared';
 
-import { IFilterVariables } from './types';
+export interface IFilterVariables {
```
Comment (ainouzgali) [inline packages/application-generic/src/utils/filter-processing-details.ts:4]: Might need to add tenant here as well. Like when we have the whole subscriber entity, we might need the tenant entity as well.

Comment (davidsoderberg) [reply packages/application-generic/src/utils/filter-processing-details.ts:4]: I do that in my next pr

**Pull request merged**